query-feature-table/src/main/java/com/esri/arcgismaps/sample/queryfeaturetable/components/MapViewModel.kt

```diff
@@ -0,0 +1,142 @@
+        FeatureLayer.createWithFeatureTable(serviceFeatureTable)
+    }
+
+    init {
+        // use symbols to show U.S. states with a black outline and yellow fill
+        val lineSymbol = SimpleLineSymbol(
+            style = SimpleLineSymbolStyle.Solid,
+            color = Color.black,
+            width = 1.0f
+        )
+        val fillSymbol = SimpleFillSymbol(
+            style = SimpleFillSymbolStyle.Solid,
+            color = Color.fromRgba(255, 255, 0, 255),
+            outline = lineSymbol
+        )
+
+        featureLayer.apply {
+            // set renderer for the feature layer
+            renderer = SimpleRenderer(fillSymbol)
+            opacity = 0.8f
+            maxScale = 10000.0
+        }
+        // add the feature layer to the map's operational layers
+        mapViewState.arcGISMap.operationalLayers.add(featureLayer)
+    }
+
+    /**
+     * Search for a U.S. state using [searchQuery] in the feature table, and if found add it
+     * to the featureLayer, zoom to it, and select it.
+     */
+    fun searchForState(searchQuery: String) {
+        // clear any previous selections
+        featureLayer.clearSelection()
+        // create a query for the state that was entered
+        val queryParameters = QueryParameters().apply {
+            // make search case insensitive
+            whereClause = ("upper(STATE_NAME) LIKE '%" + searchQuery.uppercase(Locale.US) + "%'")
+        }
+
+        sampleCoroutineScope.launch {
+            // call select features
+            val featureQueryResult = serviceFeatureTable.queryFeatures(queryParameters).getOrElse {
+                messageDialogVM.showMessageDialog(it.message.toString(), it.cause.toString())
+            } as FeatureQueryResult
+
+            val resultIterator = featureQueryResult.iterator()
+            if (resultIterator.hasNext()) {
+                resultIterator.next().run {
+                    // select the feature
+                    featureLayer.selectFeature(this)
+                    // get the extent of the first feature in the result to zoom to
+                    val envelope = geometry?.extent
+                        ?: return@launch messageDialogVM.showMessageDialog("Error retrieving geometry extent")
+                    // update the map view to set the viewpoint to the state geometry
+                    mapViewState.stateGeometry = envelope
+                }
+            } else {
+                messageDialogVM.showMessageDialog("No states found with name: $searchQuery")
+            }
+        }
+    }
+}
+
+/**
+ * Class that represents the MapView state
+ */
+class MapViewState {
+    // map used to display the feature layer
+    var arcGISMap: ArcGISMap by mutableStateOf(ArcGISMap(BasemapStyle.ArcGISTopographic))
+
+    // geometry of the queried state
+    var stateGeometry: Geometry? by mutableStateOf(null)
+
+    // set an initial viewpoint over the USA
+    var viewpoint: Viewpoint = Viewpoint(
```

Comment:
should this be a mutablestate?

Reply:
Not really, the `viewpoint` here refers to the initial viewpoint of the map and is accessed once and not modified elsewhere. I've renamed it to `initialViewpoint` for readability. 

The MapView's viewpoint is updated using `setViewpointGeometry` which is why `stateGeometry` is a mutableState.

Reply:
In that case, could it be val?